src/ci/citool/src/metrics.rs

```diff
@@ -74,6 +74,17 @@ Maybe it was newly added?"#,
 }
 
 pub fn download_job_metrics(job_name: &str, sha: &str) -> anyhow::Result<JsonRoot> {
+    // Best effort cache to speed-up local re-executions of citool
+    let cache_path = PathBuf::from(".citool-cache").join(sha).join(format!("{job_name}.json"));
+    if cache_path.is_file() {
+        if let Ok(metrics) = std::fs::read_to_string(&cache_path)
+            .map_err(|err| err.into())
```

Comment:
```suggestion
            .context("failed to read cache file")
```
provide more context to the error message

Reply:
There is no error message here :sweat_smile: The error is completely ignored, because I don't want problems with the cache to block the progress of the function. I just had to make the errors the same type, otherwise it wouldn't compile.